src/api/goods.ts

```
@@ -1,16 +1,37 @@
 import { Good } from '../types/Good';
 
-// eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAllGoods(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`Error: ${response.status}`);
+    }
+
+    return await response.json();
+  } catch (error) {
+    throw new Error('Failed to load goods');
+  }
+}
+
+export async function getFirstFiveGoods(): Promise<Good[]> {
+  try {
+    const goods = await getAllGoods();
+
+    return [...goods].sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5);
+  } catch (error) {
+    throw new Error('Failed to load 5 first goods:');
```

This custom error message hides the original error from `getAllGoods` and has a trailing colon that seems unintentional. For easier debugging, it's better to re-throw the original error (`throw error;`) or wrap it to include the original message.